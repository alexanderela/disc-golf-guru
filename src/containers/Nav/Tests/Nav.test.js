import React from 'react';
import ReactDOM from 'react-dom';
import { Nav } from '../';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router';

describe('Nav', () => {
  let wrapper;
  let mockEvent;

  beforeEach(() => {
    wrapper = shallow(<Nav location={'/findcourses'} />);
    mockEvent = { target: { name: 'home' } };
  });

  it('should render like snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
