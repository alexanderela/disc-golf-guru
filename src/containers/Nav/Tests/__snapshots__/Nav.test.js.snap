// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should render like snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav
    location="/findcourses"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="/"
        >
          <button
            className="nav-btn nav-btn-inactive"
            name="home"
          >
            Home
          </button>
        </NavLink>,
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="/findcourses"
        >
          <button
            className="nav-btn nav-btn-inactive"
            name="findCourse"
          >
            Find A Course
          </button>
        </NavLink>,
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="/favorites"
        >
          <button
            className="nav-btn nav-btn-inactive"
            name="favorites"
          >
            Favorite Courses
          </button>
        </NavLink>,
      ],
      "className": "Nav",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": <button
            className="nav-btn nav-btn-inactive"
            name="home"
          >
            Home
          </button>,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Home",
            "className": "nav-btn nav-btn-inactive",
            "name": "home",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Home",
          "type": "button",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": <button
            className="nav-btn nav-btn-inactive"
            name="findCourse"
          >
            Find A Course
          </button>,
          "to": "/findcourses",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Find A Course",
            "className": "nav-btn nav-btn-inactive",
            "name": "findCourse",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Find A Course",
          "type": "button",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": <button
            className="nav-btn nav-btn-inactive"
            name="favorites"
          >
            Favorite Courses
          </button>,
          "to": "/favorites",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Favorite Courses",
            "className": "nav-btn nav-btn-inactive",
            "name": "favorites",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Favorite Courses",
          "type": "button",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            aria-current="page"
            to="/"
          >
            <button
              className="nav-btn nav-btn-inactive"
              name="home"
            >
              Home
            </button>
          </NavLink>,
          <NavLink
            activeClassName="active"
            aria-current="page"
            to="/findcourses"
          >
            <button
              className="nav-btn nav-btn-inactive"
              name="findCourse"
            >
              Find A Course
            </button>
          </NavLink>,
          <NavLink
            activeClassName="active"
            aria-current="page"
            to="/favorites"
          >
            <button
              className="nav-btn nav-btn-inactive"
              name="favorites"
            >
              Favorite Courses
            </button>
          </NavLink>,
        ],
        "className": "Nav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": <button
              className="nav-btn nav-btn-inactive"
              name="home"
            >
              Home
            </button>,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Home",
              "className": "nav-btn nav-btn-inactive",
              "name": "home",
              "onClick": undefined,
            },
            "ref": null,
            "rendered": "Home",
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": <button
              className="nav-btn nav-btn-inactive"
              name="findCourse"
            >
              Find A Course
            </button>,
            "to": "/findcourses",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Find A Course",
              "className": "nav-btn nav-btn-inactive",
              "name": "findCourse",
              "onClick": undefined,
            },
            "ref": null,
            "rendered": "Find A Course",
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": <button
              className="nav-btn nav-btn-inactive"
              name="favorites"
            >
              Favorite Courses
            </button>,
            "to": "/favorites",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Favorite Courses",
              "className": "nav-btn nav-btn-inactive",
              "name": "favorites",
              "onClick": undefined,
            },
            "ref": null,
            "rendered": "Favorite Courses",
            "type": "button",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
